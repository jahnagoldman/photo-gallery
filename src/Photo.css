.PhotoContainer {
  background-size: 300px 150px;
  display: flex;
  justify-content: center;
}

.BlankPhoto {
  display: flex;
  justify-content: center;
  margin-top: 20px;
}

.PhotoCaption {
  font-family: Helvetica;
  font-size: 40px;
  font-weight: bold;
  position: absolute;
  top: 10%;
  left: 50%;
  pointer-events: none;
}

img {
  cursor: pointer;
  height: auto;
  margin-top: 40px;
  max-width: 90%;
}