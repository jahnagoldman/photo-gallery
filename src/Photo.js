import React from 'react';
import { PropTypes } from 'prop-types';
import './Photo.css';

function Photo(props) {
  function isPhotoMissing () {
    return !props.image  || !props.image.url;
  }

  function isCaptionMissing () {
    return !props.image || !props.image.caption;
  }
  return (
    <div className="PhotoContainer">
      { isPhotoMissing()
        ?  <div className="BlankPhoto">No photo.</div>
        : <img src={props.image.url} alt="new" onClick={props.handlePhotoClick} />
      }

      <div className="PhotoCaption">{isCaptionMissing() ? '' : props.image.caption}</div>
    </div>
  );
}

Photo.propTypes = {
  image: PropTypes.object,
  handlePhotoClick: PropTypes.func
};

export default Photo;
