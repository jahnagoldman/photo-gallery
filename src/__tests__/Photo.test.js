import React from 'react';
import { act } from 'react-dom/test-utils';
import { render, unmountComponentAtNode } from "react-dom";
import Photo from '../Photo';

describe('Photo', () => {
  let container = null;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });
      
  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
      
  it('renders the photo blank state', () => {
    act(() => {
        render(<Photo />, container);
    });

    expect(container).toMatchSnapshot();
    expect(container.textContent).toBe('No photo.');
    expect(container.querySelector('img')).toBeFalsy();
  });

  it('renders the photo with image, no caption', () => {
    act(() => {
        render(<Photo image={{'url': 'abc.com'}}/>, container);
    });

    expect(container).toMatchSnapshot();
    expect(container.textContent).toBe('');
    expect(container.querySelector('img')).toBeTruthy();
  });

  it('renders the photo with caption, no image.', () => {
    act(() => {
        render(<Photo image={{'caption': 'pic 1'}}/>, container);
    });

    expect(container).toMatchSnapshot();
    expect(container.textContent).toBe('No photo.pic 1');
    expect(container.querySelector('img')).toBeFalsy();
  });

  it('renders the photo with caption and image.', () => {
    act(() => {
        render(<Photo image={{'url': 'abc.com', 'caption': 'pic 1'}}/>, container);
    });

    expect(container).toMatchSnapshot();
    expect(container.textContent).toBe('pic 1');
    expect(container.querySelector('img')).toBeTruthy();
  });
})