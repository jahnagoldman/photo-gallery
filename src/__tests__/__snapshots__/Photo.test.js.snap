// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo renders the photo blank state 1`] = `
<div>
  <div
    class="PhotoContainer"
  >
    <div
      class="BlankPhoto"
    >
      No photo.
    </div>
    <div
      class="PhotoCaption"
    />
  </div>
</div>
`;

exports[`Photo renders the photo with caption and image. 1`] = `
<div>
  <div
    class="PhotoContainer"
  >
    <img
      alt="new"
      src="abc.com"
    />
    <div
      class="PhotoCaption"
    >
      pic 1
    </div>
  </div>
</div>
`;

exports[`Photo renders the photo with caption, no image. 1`] = `
<div>
  <div
    class="PhotoContainer"
  >
    <div
      class="BlankPhoto"
    >
      No photo.
    </div>
    <div
      class="PhotoCaption"
    >
      pic 1
    </div>
  </div>
</div>
`;

exports[`Photo renders the photo with image, no caption 1`] = `
<div>
  <div
    class="PhotoContainer"
  >
    <img
      alt="new"
      src="abc.com"
    />
    <div
      class="PhotoCaption"
    />
  </div>
</div>
`;
